lib/throttle.js

```diff
@@ -25,6 +25,21 @@ module.exports = (api) => {
     };
   };
 
+  api.metasync.debounce = (
+    timeout,
+    fn,
+    args
+  ) => {
+    let timer = null;
+    return () => {
+      clearTimeout(timer);
+      timer = setTimeout(
+        () => (args !== undefined ? fn(...args) : fn()),
+        timeout
```

Comment:
Yes, that's true. The problem is that if the user doesn't stop typing, the debounced function will never be called in your implementation, while it should be called every 500 ms, I suppose.